### PR TITLE
Update resource manager auto config and add testcases

### PR DESCRIPTION
**Pull request by moarychan**
Fixes Azure/azure-sdk-for-java#21593

Changelog:
- Fix tested issue and add resource manager test cases 
- Add azure cloud platform test cases

TODO:
The property 'spring.cloud.azure.xxx.profile.cloud=azure_china' in each service properties is not working yet.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/resourcemanager/AzureResourceManagerAutoConfigurationTest.java [hunk 1]
```diff
@@ -60,4 +65,44 @@ void testWithoutAzureResourceMetadataClass() {
...
+            .withBean(AzureResourceManager.class, AzureResourceManagerExt::getAzureResourceManager)
+            .withPropertyValues(
+                "spring.cloud.azure.profile.tenant-id=test-tenant-id",
+                "spring.cloud.azure.profile.subscription-id=test-subscription-id",
+                "spring.cloud.azure.profile.cloud=AZURE_CHINA"
+            )
+            .run(context -> {
+                assertThat(context).hasSingleBean(AzureProfile.class);
+                AzureProfile azureProfile = context.getBean(AzureProfile.class);
+                Assertions.assertEquals(azureProfile.getEnvironment().getActiveDirectoryEndpoint(),
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/resourcemanager/AzureResourceManagerAutoConfigurationTest.java:98]: I assume these two properties are not relevant to resource manager?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/resourcemanager/AzureServiceBusResourceManagerAutoConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+class AzureServiceBusResourceManagerAutoConfigurationTest {
+    private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+        .withConfiguration(AutoConfigurations.of(AzureServiceBusResourceManagerAutoConfiguration.class));
+
+    private final String connectionString = "connection-string=Endpoint=sb://test.servicebus.windows.net/;"
+        + "SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=ByyyxxxUw=";
+
+    @Test
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/resourcemanager/AzureServiceBusResourceManagerAutoConfigurationTest.java:25]: There should also be a test case where the resource manager bean be auto-configured correctly?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/resourcemanager/AzureStorageQueueResourceManagerAutoConfigurationTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import com.azure.spring.cloud.autoconfigure.storage.queue.AzureStorageQueueProperties;
+import com.azure.spring.cloud.resourcemanager.connectionstring.StorageQueueArmConnectionStringProvider;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AzureStorageQueueResourceManagerAutoConfigurationTest {
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/resourcemanager/AzureStorageQueueResourceManagerAutoConfigurationTest.java:14]: Same for this one, also same for the eh resource manager.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/resourcemanager/AzureResourceManagerAutoConfigurationTest.java (hunk 1 above)
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/resourcemanager/AzureResourceManagerAutoConfigurationTest.java:98]: You mean the `tenant-id` and `subscription-id`? it's required for AzureResourceManagerAutoConfiguration, https://github.com/Azure/azure-sdk-for-java/blob/15dda6cdc3219e9128a4d5207cb66d891fd1baf8/sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/resourcemanager/AzureResourceManagerAutoConfiguration.java#L25

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**